core/src/main/java/com/linecorp/armeria/client/endpoint/WeightedRoundRobinStrategy.java

```diff
@@ -80,16 +64,26 @@ public EndpointSelectionStrategy strategy() {
 
         @Override
         public Endpoint select() {
-            List<Endpoint> endpoints = endpointGroup.endpoints();
             long currentSequence = sequence.getAndIncrement();
 
-            if (minWeight < maxWeight) {
-                HashMap<Endpoint, AtomicInteger> endpointWeights = new LinkedHashMap<>();
-                for (Endpoint endpoint : endpoints) {
-                    endpointWeights.put(endpoint, new AtomicInteger(endpoint.weight()));
-                }
+            List<Endpoint> endpoints = endpointGroup.endpoints();
+            if (endpoints.isEmpty()) {
+                throw new EndpointGroupException(endpoints + " is empty");
+            }
+            int minWeight = Integer.MAX_VALUE;
+            int maxWeight = Integer.MIN_VALUE;
+            int totalWeight = 0;
+            Map<Endpoint, AtomicInteger> endpointWeights = new LinkedHashMap<>(endpoints.size());
```

Comment:
Having to build this on every `select()` seems too expensive. Could you add a TODO that suggests some sort of change notification mechanism?